### PR TITLE
New Feature Changes

### PR DESCRIPTION
**Pull request by milanbhagwat**
What kind of change does this PR introduce? (check one with "x")

[] Bugfix
[*] Feature
[] Code style update (formatting, local variables)
[] Refactoring (no functional changes, no api changes)
[] Other... Please describe:
Enhancement#: corresponding to commit: ee0ec2a

What is the current behavior?
Currently user do not have option to ack pulsar message need to wait for flow's successful completion.
What is the new behavior?
Now with new activity, user can configure to acknowledge message to subscriber as soon required and do need to wait for flow's completion. 

Enhancement#: corresponding to commit: 950a6fa

What is the current behavior?
Currently third party lib logger is not configurable so generating redundant logs.
What is the new behavior?
Now with new version we can configure client lib logger.
we are configuring client logger to DEBUG level when flogo log level is DEBUG. For all other log levels lib logger will be set to   ERROR level.

Enhancement#: corresponding to commit: b25c1dc

What is the current behaviour?
No functional behaviour change for Trigger
What is the new behaviour?
No functional behaviour change. Due to upgrading third party client library (pulsar-client-go), updating the struct field 'Topic' to 'DeadLetterTopic'

### CODE REVIEW

File: pulsar/connection/connection.go [hunk 1]
```diff
@@ -75,11 +77,24 @@ func (*Factory) NewManager(settings map[string]interface{}) (connection.Manager,
 		}
 	}
 
+	customLogger := plogger.NewLoggerWithLogrus(logrus.StandardLogger())
+
+	// set debug log level if FE engine configured for 'DEBUG' else ERROR level
+
+	fll := os.Getenv("FLOGO_LOG_LEVEL")
```
Comment (vnalawad-tibco) [inline pulsar/connection/connection.go]: Instead of hardcoding FLOGO_LOG_LEVEL, you should use **log.EnvKeyLogLevel**.

**Review by vnalawad-tibco — CHANGES_REQUESTED** [review]

File: pulsar/connection/connection.go [hunk 2]
```diff
@@ -75,11 +77,24 @@ func (*Factory) NewManager(settings map[string]interface{}) (connection.Manager,
 		}
 	}
 
+	customLogger := plogger.NewLoggerWithLogrus(logrus.StandardLogger())
```
Comment (lixingwang) [inline pulsar/connection/connection.go]: As we are using in flogo env,  I think it'd better have a flogo logger wrapper on this and use flogo logger.  We don't want to use logrus in flogo to have 2 different logger/formatt.

**Review by lixingwang — CHANGES_REQUESTED** [review]

File: pulsar/connection/connection.go (hunk 2 above)
Comment (lixingwang) [reply pulsar/connection/connection.go]: In that case, you don't need to get the log level and set log level.  All managed by flogo engine

Comment (milanbhagwat) [reply pulsar/connection/connection.go]: Third party's library generates redundant/informative logs at Info level and so having same logger for Flogo and third party woudn't be helpful. So we want to get more logs from thirdparty library only when user running Flogo at Debug level for debugging perspective. For Info or any other level we configuring thirdparty logger at ERROR level.

File: pulsar/connection/connection.go (hunk 1 above)
Comment (milanbhagwat) [reply pulsar/connection/connection.go]: Made corresponding changes into commit f71f7a6.

File: pulsar/connection/connection.go (hunk 2 above)
Comment (lixingwang) [reply pulsar/connection/connection.go]: If I am not wrong the lib just requires an interface of `log.Logger`.   I meant to create flogo pulsar own logger which implements log.Logger and wrapper from flogo logger.

File: pulsar/go.mod [hunk 3]
```diff
@@ -3,7 +3,8 @@ module github.com/project-flogo/messaging-contrib/pulsar
 go 1.12
 
 require (
-	github.com/apache/pulsar-client-go v0.2.0
+	github.com/apache/pulsar-client-go v0.3.0
 	github.com/apache/pulsar/pulsar-function-go v0.0.0-20200819065839-6cae4afac3c3
 	github.com/project-flogo/core v0.10.1
+	github.com/sirupsen/logrus v1.4.2
```
Comment (lixingwang) [inline pulsar/go.mod]: We wrapper Zap as pulsar client log.  I think we can remove logrus now.

**Review by lixingwang — CHANGES_REQUESTED** [review]

File: pulsar/connection/connection.go (hunk 2 above)
Comment (milanbhagwat) [reply pulsar/connection/connection.go]: updated code with zapwrapper logger. Done

File: pulsar/go.mod (hunk 3 above)
Comment (milanbhagwat) [reply pulsar/go.mod]: go.mod and sum updated with removing logrus dependency. Done

**Review by lixingwang — APPROVED** [review]
Looks good to me

**Review by vijaynalawade — APPROVED** [review]

**Pull request merged**